src/NzbDrone.Common/UniqueIdentifier/UniqueIdentifierGenerator.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using NzbDrone.Common.Cache;
+
+namespace NzbDrone.Common.UniqueIdentifier
+{
+    public interface IUniqueIdentifierGenerator
+    {
+        int Get(string key);
+    }
+
+    public class UniqueIdentifierGenerator : IUniqueIdentifierGenerator
+    {
+        private static int _lastIdentifier;
+
+        private ICached<int> _cache;
+
+        public UniqueIdentifierGenerator(ICacheManager cacheManager)
+        {
+            _cache = cacheManager.GetCache<int>(typeof(UniqueIdentifierGenerator));
+        }
+
+        public int Get(string key)
+        {
+            return _cache.Get(key, GetNextIdentifier, TimeSpan.FromHours(24));
```

Comment:
Also, one thing to keep in mind, the lib we use for caching does support expiry but doesn't actually purge items (it uses the expiry to figure out if it should refetch the values). So setting the expiry to 24 hours in this case doesn't protect you against running out of memory.